packages/core/test/unit/sql/insert.test.js

```diff
@@ -66,6 +66,70 @@ describe(Support.getTestDialectTeaser('SQL'), () => {
           bind: { sequelize_1: 0 },
         });
     });
+
+    it(
+      current.dialect.supports.inserts.onConflictWhere
+        ? 'adds conflictWhere clause to generated queries'
+        : 'throws error if conflictWhere is provided',
+      () => {
+        const User = Support.sequelize.define(
+          'user',
+          {
+            username: {
+              type: DataTypes.STRING,
+              field: 'user_name',
+              primaryKey: true,
+            },
+            password: {
+              type: DataTypes.STRING,
+              field: 'pass_word',
+            },
+            createdAt: {
+              type: DataTypes.DATE,
+              field: 'created_at',
+            },
+            updatedAt: {
+              type: DataTypes.DATE,
+              field: 'updated_at',
+            },
+          },
+          {
+            timestamps: true,
+          },
+        );
+
+        const upsertKeys = ['user_name'];
+
+        let result;
+
+        try {
+          result = sql.insertQuery(
+            User.tableName,
+            { user_name: 'testuser', pass_word: '12345' },
+            User.fieldRawAttributesMap,
+            {
+              updateOnDuplicate: ['user_name', 'pass_word', 'updated_at'],
+              conflictWhere: {
+                user_name: 'test where value',
+              },
+              upsertKeys,
+            },
+          );
+        } catch (error) {
+          result = error;
+        }
+
+        expectsql(result, {
+          default: new Error(
+            'missing dialect support for conflictWhere option',
+          ),
+          postgres:
+            'INSERT INTO "users" ("user_name","pass_word") VALUES ($sequelize_1,$sequelize_2) ON CONFLICT ("user_name") WHERE "user_name" = \'test where value\' DO UPDATE SET "user_name"=EXCLUDED."user_name","pass_word"=EXCLUDED."pass_word","updated_at"=EXCLUDED."updated_at";',
+          sqlite:
+            'INSERT INTO `users` (`user_name`,`pass_word`) VALUES ($sequelize_1,$sequelize_2) ON CONFLICT (`user_name`) WHERE `user_name` = \'test where value\' DO UPDATE SET `user_name`=EXCLUDED.`user_name`,`pass_word`=EXCLUDED.`pass_word`,`updated_at`=EXCLUDED.`updated_at`;',
```

Comment:
```suggestion
          'postgres sqlite':
            `INSERT INTO [users] ([user_name],[pass_word]) VALUES ($sequelize_1,$sequelize_2) ON CONFLICT ([user_name]) WHERE [user_name] = 'test where value' DO UPDATE SET [user_name]=EXCLUDED.[user_name],[pass_word]=EXCLUDED.[pass_word],[updated_at]=EXCLUDED.[updated_at];`,
```
A new change, but this is now possible